pkg/upgrade/upgrades/system_job_info_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package upgrades_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/server"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
+	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
+	"github.com/cockroachdb/cockroach/pkg/upgrade/upgrades"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+)
+
+func TestSystemJobInfoMigration(t *testing.T) {
+	skip.UnderStressRace(t)
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+
+	settings := cluster.MakeTestingClusterSettingsWithVersions(
+		clusterversion.TestingBinaryVersion,
+		clusterversion.ByKey(clusterversion.V22_2),
+		false,
+	)
+
+	tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			Settings: settings,
+			Knobs: base.TestingKnobs{
+				Server: &server.TestingKnobs{
+					DisableAutomaticVersionUpgrade: make(chan struct{}),
+					BinaryVersionOverride:          clusterversion.ByKey(clusterversion.V22_2),
```

Comment:
Eh, I viscerally dislike the vFoo-1 "trick": it has a the terrible effect of meaning that you can delete the preceding  versions, such that vFoo is the 0th element, and the compiler is perfectly happy, since you deleted those previous things and their usages just as you should, but then this -1 at runtime is an out-of-bound crash, since it was _implicitly_ referring to a thing you deleted, but in a way that was hidden from the compiler. I hate it, and I want to lead by example here and show that you should use actual symbol, by name, of the before version.

Reply:
Ok, can you file an issue to track the desire to adopt that pattern throughout all upgrades, please? Seems worth it.

Reply:
https://github.com/cockroachdb/cockroach/issues/91675